src/DSharp.Compiler/Compiler/ExpressionBuilder.cs

```diff
@@ -922,6 +975,21 @@ MethodExpression methodExpression
             return expression;
         }
 
+        private Expression CreateExtensionMethodInvocationExpression(BinaryExpressionNode node, TypeSymbol typeToExtend)
+        {
+            string memberName = ((NameNode)node.RightChild).Name;
+            MethodSymbol methodSymbol = symbolSet.ResolveExtensionMethodSymbol(typeToExtend.FullName, memberName);
+
+            MethodExpression methodExpression = new MethodExpression(
+                        new TypeExpression((TypeSymbol)methodSymbol.Parent, SymbolFilter.Public | SymbolFilter.StaticMembers),
+                        methodSymbol);
+            Expression accessorExpression = BuildExpression(node.LeftChild);
+            methodExpression.AddParameterValue(accessorExpression);
+            methodExpression.IsExtensionMethod = true;
```

Comment:
if we mark it as static, the reversing of the nodes (left into right) happens automatically

Reply:
Not sure I follow what you mean, Mark which bit as static? And what reverses? :D

Reply:
In this case I need to know if a method is static so that i can short circuit out in the next example.